### PR TITLE
Bug 1216183 - [Messages] We should not fail when the attachment is in correct

### PR DESCRIPTION
**Pull request by kumarrishav**

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=4307ab35f35a2af67cc5fa832fb8d68b8c40a352

File: apps/sms/views/conversation/js/attachment.js [hunk 1]
```diff
@@ -26,7 +26,8 @@
   function Attachment(blob, options) {
     options = options || {};
     this.blob = blob;
-    this.name = blob.name || options.name || '';
+    this.name = typeof blob.name == 'string' ? blob.name : (
+      typeof options.name == 'string' ? options.name : '');
```
Comment (julienw) [inline apps/sms/views/conversation/js/attachment.js]: maybe it's more readable to keep the previous line, and add a new one `if (typeof this.name !== 'string') { this.name = '' }`

What do you think ?

File: apps/sms/views/conversation/js/compose.js [hunk 2]
```diff
@@ -679,7 +679,7 @@ var Compose = (function() {
 
       var currentAttachmentDOM = event.target;
       var currentAttachment = attachments.get(event.target);
-      var name = currentAttachment.name;
+      var name = currentAttachment.name || '';
```
Comment (julienw) [inline apps/sms/views/conversation/js/compose.js]: it's likely not necessary with the code we have in attachment.js :) I know this was my first proposal but this was before I properly identified the issue.

File: apps/sms/views/conversation/js/attachment.js (hunk 1 above)
Comment (julienw) [inline apps/sms/views/conversation/js/attachment.js]: also, I'd appreciate unit tests ;)

Comment (kumarrishav) [inline apps/sms/views/conversation/js/attachment.js]: @julienw  ah ah. your code is so clean like alwz :smiley: 
if (typeof this.name !== 'string') { this.name = '' } -> okay with it

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=b22d9d5847f74ddc348db52ae0e4dd9499ee2e74

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=1edeb554ad1a910d4e0defa8906febe73bfafbbe

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=e9e9b520013d0a181a84429985722032823ab005

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=0e7483bc2ab640b7b25607c0f1870b701fdf33de

File: apps/sms/views/conversation/js/attachment.js [hunk 3]
```diff
@@ -23,11 +25,19 @@
...
+  * @param {Object, Object} (blob object, options object)
+  */
   function Attachment(blob, options) {
     options = options || {};
     this.blob = blob;
-    this.name = blob.name || options.name || '';
     this.isDraft = !!options.isDraft;
+    this.name = blob.name || options.name || '';
+
+    if(typeof this.name !== 'string') {
```
Comment (julienw) [inline apps/sms/views/conversation/js/attachment.js]: nit: please add a space after "if"

File: apps/sms/views/conversation/test/unit/attachment_test.js [hunk 4]
```diff
@@ -51,6 +51,14 @@ suite('attachment_test.js', function() {
     assert.typeOf(attachment.name, 'string');
   });
 
+  test('Filename is not a string', function() {
+    var attachment = new Attachment(testImageBlob, {
+      name: 1111
```
Comment (julienw) [inline apps/sms/views/conversation/test/unit/attachment_test.js]: nit: maybe use an object like `{}` ?

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=0febad39d1c3e3d907eb435a58acd42e20034ab9

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=93905aa12f71d5f3e37398cf6a9c0924e63a9ab0

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=e0a826cae0ea43b59386aef92b4381e5c1cb38af

**Pull request merged**